src/hotspot/share/gc/serial/markSweep.cpp

```diff
@@ -252,4 +242,10 @@ void MarkSweep::initialize() {
   MarkSweep::_gc_timer = new STWGCTimer();
   MarkSweep::_gc_tracer = new SerialOldTracer();
   MarkSweep::_string_dedup_requests = new StringDedup::Requests();
+
+  static AlwaysTrueClosure always_true;
```

Comment:
Please put this static into the class as static class member; it's very hard to track if statics are put into random places.
Why we can use `AlwaysTrueClosure` here should probably be documented.
The reference processor for `MarkSweep` can also be a static, as well as `MarkAndPushClosure` initialized statically with that reference processor.

Reply:
> Please put this static into the class as static class member; it's very hard to track if statics are put into random places. 

This refers to the `AlwaysTrueClosure`.

>Why we can use `AlwaysTrueClosure` here should probably be documented.

Something like: "Mark&Sweep collection covers the whole heap so all objects should be subject to discovery"

>The reference processor for `MarkSweep` can also be a static, as well as `MarkAndPushClosure` initialized statically with that reference processor.

Here I ommitted the important part: `_ref_processor` does not need to be a reference/pointer any more and we can get rid of the new/delete (I think).

Reply:
Directly embedding `ReferenceProcessor` as a field doesn't work, because its constructor uses static variables (e.g. `_default_soft_ref_policy`) that are lazily initialized.